### PR TITLE
Feature/update to groovy 4.x

### PR DESCRIPTION
**Pull request by royteeuwen**

### CODE REVIEW

**Review by bdhoine — APPROVED** [review]

**Comment by royteeuwen** [comment]
Waiting for https://github.com/apache/groovy/pull/1836 to be merged and released. groovy-json is not working as expected in OSGi

File: groovy/groovy-json/pom.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        <artifactId>aem-groovy-console</artifactId>
+        <version>18.0.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>groovy-json</artifactId>
+    <name>Groovy JSON</name>
+    <packaging>bundle</packaging>
+    <version>${groovy.version}.1-FIX</version>
+
+    <description>Groovy JSON Fix to make it compatible with OSGi until GROOVY-10881 is fixed</description>
```
Comment (bdhoine) [inline groovy/groovy-json/pom.xml:19]:
```suggestion
    <description>Groovy JSON Fix to make it compatible with OSGi until https://issues.apache.org/jira/browse/GROOVY-10881 is fixed</description>
```

File: groovy/groovy-json/src/main/java/org/apache/groovy/json/internal/FastStringUtils.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package org.apache.groovy.json.internal;
+
+import org.apache.groovy.json.DefaultFastStringService;
+import org.apache.groovy.json.FastStringService;
+
+/**
+ * To be removed once GROOVY-10881 is fixed
```
Comment (bdhoine) [inline groovy/groovy-json/src/main/java/org/apache/groovy/json/internal/FastStringUtils.java:7]:
```suggestion
 * FIXME: To be removed once GROOVY-10881 is fixed
```

File: pom.xml [hunk 3]
```diff
@@ -87,8 +88,10 @@
         <vault.password>admin</vault.password>
         <package.group>aem-groovy-console</package.group>
 
-        <!-- versions -->
-        <aem.sdk.version>2022.10.9398.20221020T071514Z-220800</aem.sdk.version>
+        <!-- AEM minimum version -->
+        <aem.version>6.5.10</aem.version>
```
Comment (bdhoine) [inline pom.xml:92]: maybe good idea to have profile for latest SDK and minimum on prem?

File: pom.xml [hunk 4]
```diff
@@ -435,11 +461,10 @@
 
             <!-- SDK included last to allow other dependencies to override included versions -->
             <dependency>
-                <groupId>io.wcm.maven</groupId>
```
Comment (bdhoine) [inline pom.xml:438]: put in profile and compile for both?

File: ui.apps/src/main/content/jcr_root/apps/groovyconsole/clientlibs/js/audit.js [hunk 5]
```diff
@@ -9,6 +9,7 @@ GroovyConsole.Audit = function () {
     return {
         initialize: function () {
             table = $('.audit').DataTable({
+                retrieve: true,
```
Comment (bdhoine) [inline ui.apps/src/main/content/jcr_root/apps/groovyconsole/clientlibs/js/audit.js:12]: merge in main branch to remove from PR?

**Review by bdhoine — COMMENTED** [review]
Non blocking comments, still need to test locally

File: README.md [hunk 6]
```diff
@@ -17,7 +17,9 @@ scripts are included in the package for reference.
 
 ## Compatibility
 
-We only support AEM as a Cloud Service. For AEM on premises, check the original [aem-groovy-console](https://github.com/CID15/aem-groovy-console)
+We only support AEM 6.5.10+ and AEM as a Cloud Service. 
+
+For older AEM on premises, check the original repo [aem-groovy-console](https://github.com/CID15/aem-groovy-console)
```
[the remark this answers is not in the document]
Comment (royteeuwen) [reply README.md:22]: done in other pr

File: README.md [hunk 7]
```diff
@@ -17,7 +17,9 @@ scripts are included in the package for reference.
 
 ## Compatibility
 
-We only support AEM as a Cloud Service. For AEM on premises, check the original [aem-groovy-console](https://github.com/CID15/aem-groovy-console)
+We only support AEM 6.5.10+ and AEM as a Cloud Service.
```
[the remark this answers is not in the document]
Comment (royteeuwen) [reply README.md:20]: done in other pr

**Pull request closed without merging**